### PR TITLE
Add password variable

### PR DESCRIPTION
**Pull request by hdep**
As stated in #16  this PR add a password variable into nsclient class and define the value into the template.
here is official documentation about this optional parameter : 

http://docs.nsclient.org/reference/check/CheckMKServer/#password

this need to be tested.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -64,14 +67,16 @@
   $package_name            = $nsclient::params::package_name,
   $download_destination    = $nsclient::params::download_destination,
   $config_template         = $nsclient::params::config_template,
-  $install_path            = $nsclient::params::install_path
+  $install_path            = $nsclient::params::install_path,
+  $password                = $nsclient::params::password
```
Comment (bastelfreak) [inline manifests/init.pp]: Clean you please add a proper datatype to the new parameter? A quote from our [docs](https://voxpupuli.org/docs/#reviewing-a-module-pr):

> Are new parameters introduced? They must have datatypes

> Are datatypes from stdlib used? Ensure that lowest supported stdlib version is 4.13.1. Check if a newer version introduced the used datatype

File: manifests/init.pp [hunk 2]
```diff
@@ -64,14 +67,16 @@
...
 ) inherits nsclient::params {
 
   validate_string($package_source_location)
   validate_string($package_source)
   validate_string($package_name)
   validate_string($config_template)
   validate_string($install_path)
+  validate_string($password)
```
Comment (bastelfreak) [inline manifests/init.pp]: Please do not use the legacy validate_*() functions. Instead we can rely on datatypes.

File: templates/nsclient.ini.erb [hunk 3]
```diff
@@ -44,6 +44,11 @@ NSClientServer = 1
 ; ALLOWED HOSTS - A comaseparated list of allowed hosts. You can use netmasks (/ syntax) or * to create ranges.
 allowed hosts = <%= @allowed_hosts.join(",") %>
 
+<% unless @password.nil? -%>
```
Comment (bastelfreak) [inline templates/nsclient.ini.erb]: This should work as well and is shorter:

```suggestion
<% if @password -%>
```

**Comment by bastelfreak** [comment]
Hi @hdep, thanks for the PR! Can you take a look at the inline comments I made and also add an unit test to verify that this works?

**Comment by hdep** [comment]
I think code is good now.  I will try to create a unit test for this.

**Comment by bastelfreak** [comment]
I updated the code and will merge it when it turns green.

File: manifests/service.pp [hunk 4]
```diff
@@ -12,9 +12,12 @@
   $service_enable  = $nsclient::service_enable,
   $allowed_hosts   = $nsclient::allowed_hosts,
   $config_template = $nsclient::config_template,
-  $install_path    = $nsclient::install_path
+  $install_path    = $nsclient::install_path,
+  $password        = $nsclient::password
```
Comment (bastelfreak) [inline manifests/service.pp]: the tests itself were okay. The variable `$password` was just unknown within the scope of the template. The template is called in the class nsclient::service, so the variable has to exist here too. I added it as a new class parameter.

**Comment by hdep** [comment]
Thank you  !

**Pull request merged**